bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,342 @@
+            if data['author'] == message.author:  # if the author has already added an entry
+                await ctx.send(
+                    "But you have already added an entry! Type "
+                    f"`{self.bot.command_prefix}spooknamerate "
+                    "delete` to delete it, and then you can add it again"
+                )
+                return
+
+            elif data['word'] == word:
+                # Someone has added the word
+                await ctx.send("TOO LATE. Someone has already added this word.")
+                return
+
+        # otherwise
+        self.messages[message.id] = {  # store it
+            'word': word,
+            'author': message.author,
+            'score': 0,
+        }
+
+        for emoji in self.emojis_val:
+            await message.add_reaction(emoji)  # add each emoji
+
+        logger.info(f"{message.author} added the word {word!r}")
+
+        await ctx.send(f"{word!r} added successfully!\n{random.choice(self.added_messages)}")
+        return
+
+    @commands.Cog.listener()
+    async def on_reaction_add(self, reaction: discord.Reaction, user: discord.User) -> None:
+        """Ensures that each user adds one and only one reaction."""
+        try:
+            if reaction.emoji in self.emojis_val and reaction.message.id in self.messages:
+                msg = reaction.message
+                msg: Message
+                # create a custom counter
+                reaction_counter = {}
+                for r in msg.reactions:
+                    async for user in r.users():
+                        if user in reaction_counter:
+                            reaction_counter[user] += 1
+                        else:
+                            reaction_counter[user] = 1
+
+                if reaction_counter[user] > 1 and user != self.bot.user:
+                    await user.send("Sorry, you have already added a reaction, please remove your reaction and try "
+                                    "again")
+                    return await reaction.remove(user)  # removes user's reaction
+        except RuntimeError:  # The dictionary was changed in between
+            pass
+
+    @spook_name_rate.command(name='delete')
+    async def delete_word(self, ctx: Context) -> None:
+        """Delete's the user's word."""
+        if self.poll:
+            await ctx.send("You can't delete your word since the poll has already started!")
+            return
+        for message_id in self.messages:
+            data = self.messages[message_id]
+            if ctx.author == data['author']:
+                del self.messages[message_id]
+                if self.messages.get(message_id) is None:  # Make sure it does not exist
+                    await ctx.send(f'Message deleted successfully ({data["word"]!r})!')
+                    return
+                else:
+                    await ctx.send("Oops, there was some error, please try again!")
+                    return
```

Comment:
> # Make sure it does not exist

Pretty sure this should never happen. It will error if you try to delete something that doesn't exist but that's about it.